a10_octavia/tests/unit/controller/worker/tasks/test_handler_virtual_server.py

```diff
@@ -1,20 +1,40 @@
+#    Copyright 2020, A10 Networks
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from unittest.mock import patch
 import mock
-import uuid
+from a10_octavia.common.data_models import VThunder
+from a10_octavia.controller.worker.tasks.handler_virtual_server import CreateVirtualServerTask
+from a10_octavia.tests.common import a10constants
+from octavia.common import data_models as o_data_models
+from octavia.tests.common import constants as t_constants
+from octavia.tests.unit import base
 
-from a10_octavia.controller.worker.tasks.handler_virtual_server import CreateVitualServerTask, DeleteVitualServerTask
-from a10_octavia.tests.test_base import TestBase
+AMPHORA = o_data_models.Amphora(id=t_constants.MOCK_AMP_ID1)
+VTHUNDER = VThunder()
+LB = o_data_models.LoadBalancer(id=a10constants.MOCK_LOAD_BALANCER_ID, amphorae=[AMPHORA])
 
-USERNAME = "user"
-PASSWORD = "pass"
+class TestCreateVirtualServerTask(base.TestCase):
 
-class TestCreateVirtualServerTask(TestBase):
     def setUp(self):
+        patcher = patch(
+            'a10_octavia.controller.worker.tasks.common.BaseVThunderTask.client_factory')
+        self.client_factory_mock = patcher.start()
+        self.client_factory_mock.return_value = mock.Mock()
```

Comment:
Instead we could do

```python
mock_patch = patcher.start()
self.client_mock = mock.Mock()
mock_patch.return_value = self.client_mock
```

Then later on we just have to check ` self.client_mock.slb.virtual_server.delete.assert_called_with(LB.id)`

Reply:
Better yet done in base class. Since everyone of our task tests uses this